plugins/jobtype/src/azkaban/jobtype/HadoopSecureSparkWrapper.java

```diff
@@ -35,25 +43,41 @@
 /**
  * <pre>
  * A Spark wrapper (more specifically a spark-submit wrapper) that works with Azkaban.
- * This class will receive input from {@link azkaban.jobtype.HadoopSparkJob}, and pass it on to spark-submit
+ * This class will be running on a separate process with JVM/ENV properties, classpath and main args
+ *  built from {@link azkaban.jobtype.HadoopSparkJob}.
+ * This class's main() will receive input args built from {@link azkaban.jobtype.HadoopSparkJob},
+ *  and pass it on to spark-submit to launch spark job.
+ * This process will be the client of the spark job.
+ *
  * </pre>
  *
  * @see azkaban.jobtype.HadoopSecureSparkWrapper
  */
 public class HadoopSecureSparkWrapper {
 
   private static final Logger logger = Logger.getRootLogger();
+  private static final String EMPTY_STRING = "";
+
+  //SPARK CONF PARAM
+  private static final String SPARK_CONF_EXTRA_DRIVER_OPTIONS = "spark.driver.extraJavaOptions";
```

Comment:
I would argue that it is a better trade off to tolerate some inconsistency during the transition period as long as we all agree that the suggestion makes sense in the long run.

It's fine to do it in a separate change though.